openadr-client/src/lib.rs

```diff
@@ -482,4 +503,136 @@ impl Client {
             program,
         ))
     }
+
+    /// Create a new event on the VTN
+    pub async fn create_event(&self, event_data: EventContent) -> Result<EventClient> {
+        let event = self.client_ref.post("events", &event_data, &[]).await?;
+        Ok(EventClient::from_event(self.client_ref.clone(), event))
+    }
```

Comment:
that complicates things because the program id of the event might not match the program id of the program client?

Reply:
oh I did not think about that, I guess technically if you create an `event` through a `ProgramClient` you should not even be able to set any program id since it is implicit... Don't know if it is worth the effort to create a whole new struct for that?